state/protocol/badger/mutator.go

```diff
@@ -700,38 +725,28 @@ func (m *FollowerState) epochTransitionMetricsAndEventsOnBlockFinalized(block *f
 	err error,
 ) {
 
-	parentBlocksEpoch := m.AtBlockID(block.ParentID).Epochs().Current()
-	parentEpochFinalView, err := parentBlocksEpoch.FinalView()
+	isFirstBlockOfEpoch, err := m.isFirstBlockOfEpoch(block)
 	if err != nil {
-		return nil, nil, fmt.Errorf("could not get parent epoch final view: %w", err)
+		return nil, nil, fmt.Errorf("could not check if finalized block is first of epoch")
+	}
+	if !isFirstBlockOfEpoch {
```

Comment:
can we move this check outside so we avoid calling `isFirstBlockOfEpoch` twice?

Reply:
+1 

I think it would be useful to rename this method to `epochTransitionMetricsOnFinalizedEpochTransition` and explicitly make it part of the API contract that this is only to be called on the first block of an epoch.

Reply:
I've moved this check into the caller and updated the requirements for the function 👍  https://github.com/onflow/flow-go/pull/3903/commits/d26b84120af20917409a127e0873ca43107b32e3